pkg/registry/client.go

```diff
@@ -459,7 +459,7 @@ func (c *Client) Pull(ref string, options ...PullOption) (*PullResult, error) {
 			PreCopy: func(_ context.Context, desc ocispec.Descriptor) error {
 				mediaType := desc.MediaType
 				if i := sort.SearchStrings(allowedMediaTypes, mediaType); i >= len(allowedMediaTypes) || allowedMediaTypes[i] != mediaType {
-					return errors.Errorf("media type %q is not allowed, found in descriptor with digest: %q", mediaType, desc.Digest)
+					return nil
```

Comment:
From ORAS code/docs:
```
        // PreCopy handles the current descriptor before it is copied. PreCopy can
	// return a SkipNode to signal that desc should be skipped when it already
	// exists in the target.
	PreCopy func(ctx context.Context, desc ocispec.Descriptor) error
```

Returning nil will cause (iiuc) ORAS to copy the layer. Which Helm is not expecting, nor wants/needs to occur. I think we want to return `SkipNode` instead, to avoid the unneeded (e.g. provenance layer being copied)

@TerryHowe  -- that sound correct?

```suggestion
					return oras.SkipNode
```

Reply:
@gjenkins8 you are correct. we do want `oras.SkipNode` as the resolution here as this type of functionality was how it was handled prior to the incorporation of ORAS v2. the ORAS library took care of it when `oras.WithAllowedMediaTypes` was provided